src/Rest/Base.php

```diff
@@ -178,7 +178,6 @@ protected static function request(
 
         $client = new Rest($session->getShop(), $session->getAccessToken());
 
-        $params = array_filter($params);
```

Comment:
I think we do want to filter out empty strings. 

Consider the following test case.

```
    public function testFindsWithInvalidParam()
    {
        $body = ["fake_resource" => ["id" => 1, "attribute" => "attribute"]];

        $this->mockTransportRequests([
            new MockRequest(
                $this->buildMockHttpResponse(200, $body),
                "{$this->prefix}/fake_resources/1.json?param2=value",
                "GET",
                null,
                ["X-Shopify-Access-Token: access-token"],
            ),
        ]);

        $resource = FakeResource::find($this->session, 1, ["param" => "", "param2" => "value"]);
        $this->assertEquals([1, "attribute"], [$resource->id, $resource->attribute]);
    }
   ```
   
With the current functionality we would build a url like `fake_resources/1.json?param2=value`. If we remove the `array_filter` we get an invalid url `fake_resources/1.json?param=&param2=value`.

I think we want to only filter out empty strings.